packages/menu/src/Menu.ts

```diff
@@ -119,18 +119,26 @@ export class Menu extends SpectrumElement {
     }
 
     public focusMenuItemByOffset(offset: number): void {
+        const step = offset || 1;
         const focusedItem = this.menuItems[this.focusedItemIndex] as MenuItem;
         focusedItem.focused = false;
         this.focusedItemIndex =
             (this.menuItems.length + this.focusedItemIndex + offset) %
             this.menuItems.length;
         let itemToFocus = this.menuItems[this.focusedItemIndex] as MenuItem;
-        while (itemToFocus.disabled) {
+        let availableItems = this.menuItems.length;
+        // cycle through the available items in the directions of the offset to find the next non-disabled item
+        while (itemToFocus.disabled && availableItems) {
+            availableItems -= 1;
             this.focusedItemIndex =
-                (this.menuItems.length + this.focusedItemIndex + offset) %
+                (this.menuItems.length + this.focusedItemIndex + step) %
                 this.menuItems.length;
             itemToFocus = this.menuItems[this.focusedItemIndex] as MenuItem;
         }
+        // if there are no non-disabled items, skip the work to focus a child
+        if (itemToFocus.disabled) {
+            return;
```

Comment:
do you need to do anything to `this.focusedItemIndex` here? I think the other places it's used are safe, but I'm not positive about all the code. I'd at least make sure the places it's used are covered in the all-disabled tests.

Reply:
Interesting...
```
- item /// focusedItemIndex, because it's the only focusable thing or because === 0 by default
- item[disabled]
- item[disabled]
```
`focusMenuItemByOffset(-1 | 0 | 1)` In this case it'll loop and rest `focusedItemIndex` to what it was as it actually focused the first item.
```
- item[disabled] /// focusedItemIndex === 0 by default
- item[disabled]
- item[disabled]
```
`focusMenuItemByOffset(0)` the case that I think Charles ran into will cause `focusedItemIndex` to start at 0 loop back around to 0 and then do nothing.

Can you share a little more about what you'd like to see in expanded tests here?